Formula/corectl.rb

```diff
@@ -13,26 +13,64 @@ class Corectl < Formula
 
   depends_on "go" => :build
   depends_on "godep" => :build
+  depends_on "ocaml" => :build
+  depends_on "opam" => :build
   depends_on :macos => :yosemite
 
+  # until 0.7.1 is out
+  if build.stable?
+    patch do
+      url "https://github.com/TheNewNormal/corectl/commit/e519a6f0d1d3c141baef631bdfc65b4e130ff60d.patch"
+      sha256 "966caa8a8f0703fffb5befcc3bec546699c06ca9512d6b1739f28029fe27ed89"
+    end
+  end
+
   def install
     ENV["GOPATH"] = buildpath
 
+    opamroot = buildpath/"opamroot"
+    opamroot.mkpath
+    ENV["OPAMROOT"] = opamroot
+    ENV["OPAMYES"] = "1"
+
     path = buildpath/"src/github.com/TheNewNormal/#{name}"
     path.install Dir["*"]
 
     args = []
     args << "VERSION=#{version}" if build.stable?
 
     cd path do
-      system "make", "corectl", *args
-      system "make", "documentation/man"
-      bin.install "corectl"
+      system "make", "default", *args
+
+      system "opam", "init", "--no-setup"
+      system "opam", "pin", "add", "qcow-format",
+        "git://github.com/mirage/ocaml-qcow#master"
+      system "opam", "install", "uri", "qcow-format", "ocamlfind"
+      system "make", "hyperkit"
+
+      bin.install "bin/corectl"
+      bin.install "bin/corectld"
+      bin.install "bin/corectld.runner"
+
       man1.install Dir["documentation/man/*.1"]
-      share.install "cloud-init", "profiles"
+      share.install "examples/cloud-init", "examples/profiles"
     end
   end
 
+  def caveats; <<-EOS.undent
+      Starting with 0.7 "corectl" has a client/server architecture. Accordingly before start
```

Comment:
This is indented a bit too much. Should be two spaces instead of four spaces/1 tab. We're fairly strict on not carrying caveats that aren't related to Homebrew specifically. We don't really want caveats to become an extension of the upstream documentation.